strictdoc/core/file_traceability_index.py

```diff
@@ -608,15 +609,50 @@ def validate_and_resolve(
                     continue
 
                 assert source_node_.entity_name is not None
+                sdoc_node = None
                 sdoc_node_uid = source_node_.get_sdoc_field(
                     "UID", relevant_source_node_entry
                 )
-                if sdoc_node_uid is None:
-                    sdoc_node_uid = f"{document_uid}/{path_to_source_file_}/{source_node_.entity_name}"
-                sdoc_node = traceability_index.get_node_by_uid_weak(
-                    sdoc_node_uid
+                mid = source_node_.get_sdoc_field(
+                    "MID", relevant_source_node_entry
                 )
 
+                # First merge criterion: Merge if SDoc node with same MID exists.
+                if mid is not None:
+                    sdoc_node_mid = MID(mid)
+                    merge_candidate_sdoc_node = (
+                        traceability_index.get_node_by_mid_weak(sdoc_node_mid)
+                    )
+                    if merge_candidate_sdoc_node is not None:
+                        merge_candidate_sdoc_node = assert_cast(
+                            merge_candidate_sdoc_node, SDocNode
+                        )
+                        # Simplification: Either SDoc must have a static UID that is not overwritten by source node.
```

Comment:
I am not sure I quite get what this comment is referring to. Could you rewrite it without `Either SDoc`?

Reply:
The comment was only half a sentence. The place is refactored now and the comment is removed. Hope new structure and comments in itests make Intention more clear.